generator/osm_source.hpp

```diff
@@ -48,6 +51,17 @@ class EmitterBase
   virtual void GetNames(vector<string> & names) const = 0;
 };
 
+class SyncOfstream
```

Comment:
Yep, seems that PARALLEL_POLYGONIZER is set here. But in any case, you may just collect pass to EmitFeature some kind of holder instead of sync ofstream, i.e. map from feature id to a vector of osm ids. And, at the end, just dump one or two maps to the file. No need in SyncOfstream.